src/lib.rs

```diff
@@ -396,27 +395,43 @@ pub fn init_logging(opentelemetry_url: &Option<Url>) -> LemmyResult<()> {
     .trim_matches('"')
     .parse::<Targets>()?;
 
-  let format_layer = {
-    #[cfg(feature = "json-log")]
-    let layer = tracing_subscriber::fmt::layer().with_ansi(false).json();
-    #[cfg(not(feature = "json-log"))]
-    let layer = tracing_subscriber::fmt::layer().with_ansi(false);
-
-    layer.with_filter(targets.clone())
-  };
-
-  let subscriber = Registry::default()
-    .with(format_layer)
-    .with(ErrorLayer::default());
-
-  if let Some(_url) = opentelemetry_url {
-    #[cfg(feature = "console")]
-    telemetry::init_tracing(_url.as_ref(), subscriber, targets)?;
-    #[cfg(not(feature = "console"))]
-    tracing::error!("Feature `console` must be enabled for opentelemetry tracing");
+  if settings.json_logging {
+    let format_layer = {
+      let layer = tracing_subscriber::fmt::layer().with_ansi(false).json();
+      layer.with_filter(targets.clone())
+    };
```

Comment:
Only this part needs to be inside the if, something like this:
```rust
let layer = if settings.json_logging {
    tracing_subscriber::fmt::layer().with_ansi(false).json();
} else {
    tracing_subscriber::fmt::layer().with_ansi(false).json();
}
let format_layer = layer.with_filter(targets.clone());
...
```

Then its not necessary to write all the stuff for opentelemetry etc twice.

Reply:
Youre right, it seems there is no other option in this case.